<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.361">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sophie D., François M., Karen P.">

<title>Vitesse de synthèse</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Cinetique_Proteines_files/libs/clipboard/clipboard.min.js"></script>
<script src="Cinetique_Proteines_files/libs/quarto-html/quarto.js"></script>
<script src="Cinetique_Proteines_files/libs/quarto-html/popper.min.js"></script>
<script src="Cinetique_Proteines_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Cinetique_Proteines_files/libs/quarto-html/anchor.min.js"></script>
<link href="Cinetique_Proteines_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Cinetique_Proteines_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Cinetique_Proteines_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Cinetique_Proteines_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Cinetique_Proteines_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Vitesse de synthèse</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Sophie D., François M., Karen P. </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<section id="contexte" class="level2">
<h2 class="anchored" data-anchor-id="contexte">Contexte</h2>
<p>On s’intéresse à des images composées de points verts et rouges, chaque point correspondant à un bout précis de l’ADN.</p>
<p>Quand ces bouts d’ADN ont été lus, la particule s’éteint.</p>
<p>Cependant, ces particuliers s’éteignent aussi même sans avoir été lus (parce que éclairés trop longtemps par exemple).</p>
<p>On note <span class="math inline">\(i\)</span> un bout d’ARN représenté par un point vert et un point rouge. Soit <span class="math inline">\(n\)</span> leur nombre.</p>
<p>Après traitement d’image, on peut observer sur un intervalle de temps donné <span class="math inline">\([T_0, T]\)</span>, les instants de mort <span class="math display">\[(T_i^R, T^V_i)_{i=1,\dots n}.\]</span></p>
<p>Deux processus de morts sont à l’oeuvre et se cumulent:</p>
<ul>
<li><p>Mort naturelle (dûe à la durée de l’éclairage)</p></li>
<li><p>Mort par lecture</p></li>
</ul>
<p>Pour toute entité <span class="math inline">\(i\)</span>, soit <span class="math inline">\(U_i^{R}\)</span> l’instant de mort naturelle de la particule <span class="math inline">\(R\)</span> (idem pour <span class="math inline">\(V\)</span>) et soit <span class="math inline">\(V_i^{R}\)</span> sa mort par lecture.</p>
<p>Alors</p>
<span class="math display">\[\begin{aligned}
T_i^{R} &amp;= \min(U_i^{R},V_i^R)\\
T_i^{V} &amp;= \min(U_i^{V},V_i^V)
\end{aligned}\]</span>
<p>De plus, par lecture le long du ribosome, la verte s’éteint avant la rouge.</p>
<p><span class="math display">\[ V_i^R &gt; V_i^V\]</span></p>
</section>
<section id="données" class="level2">
<h2 class="anchored" data-anchor-id="données">Données</h2>
<p>On observe deux types de données.</p>
<p>-Expérience 1 : données de contrôle permettent d’observer le phénomène de mort naturelle <span class="math inline">\((T_i^{C})_{i=1, \dots,n_C}\)</span>.</p>
<p>-Expérience 2 : expérience de lecture <span class="math inline">\((T_i^{R}, T_i^V)_{i=1, \dots,n}\)</span>.</p>
</section>
<section id="le-modèle" class="level2">
<h2 class="anchored" data-anchor-id="le-modèle">Le modèle</h2>
<p><span class="math display">\[
\left\{
\begin{array}{cclcl}
U_i^R &amp;\sim&amp;   \mathcal{F}(\theta_{N}) &amp;=&amp; \Gamma(\alpha_N,\beta_N)\\
U_i^V &amp;\sim&amp;   \mathcal{F}(\theta_{N}) &amp;=&amp; \Gamma(\alpha_N,\beta_N) \\
V_i^V  &amp;\sim&amp;    \mathcal{F}(\theta_{L}) &amp;=&amp; \mathcal{E}(\lambda_c) + \Gamma(k, \lambda_e)  \\
V_i^R  &amp;\sim&amp;  V_i^V + \mathcal{F}(\nu_{L}) &amp;=&amp; V_i^V + \Gamma(k', \lambda_e)  \\
T_i^{R} &amp;=&amp;  \min(U_i^{R},V_i^R)\\
T_i^{V} &amp;=&amp; \min(U_i^{V},V_i^V)
\end{array}
\right.
\]</span></p>
<p><span class="math inline">\(\alpha_N\)</span> et <span class="math inline">\(\beta_N\)</span> sont les paramètres de mort spontanée. <span class="math inline">\(\lambda_c\)</span> représente le temps pour que la particule de rejoindre la protéine. <span class="math inline">\(k\)</span> est le nombre de codons. <span class="math inline">\(\lambda_e\)</span> est le paramètre de lecture pour chaque codon (???). <span class="math inline">\(k'\)</span> est le nombre de codons (???) entre le début et la fin de la séquence.</p>
</section>
<section id="estimation-des-paramètres-de-mort-spontanée." class="level2">
<h2 class="anchored" data-anchor-id="estimation-des-paramètres-de-mort-spontanée.">Estimation des paramètres de mort spontanée.</h2>
<p>Ces paramètres peuvent être estimés à partir des données de l’expérience dite de contrôle. Les temps <span class="math inline">\((T_i^C)_{i=1,\dots,c_C}\)</span> suivent le modèle suivant:</p>
<p><span class="math display">\[
\begin{array}{ccl}
T_i^{C} &amp;\sim&amp; \Gamma(\alpha_N, \beta_N)
\end{array}
\]</span></p>
<p>Par conséquent, les paramètres <span class="math inline">\(\alpha_N\)</span> et <span class="math inline">\(\beta_N\)</span> peuvent être estimés par les estimateurs des moments.</p>
<p><span class="math display">\[\widehat{\alpha}_N = \frac{\overline{T^C}^2}{Var(T^C)}, \quad \widehat{\beta}_N = \frac{\widehat{\alpha}_N}{\overline{T^C}}\]</span></p>
</section>
<section id="estimation-des-paramètres-de-synthèse." class="level2">
<h2 class="anchored" data-anchor-id="estimation-des-paramètres-de-synthèse.">Estimation des paramètres de synthèse.</h2>
<p>On cherche à estimer <span class="math inline">\(\lambda_e\)</span> et <span class="math inline">\(\lambda_c\)</span>. On fixe <span class="math inline">\(\hat{\theta}_N = (\widehat{\alpha}_N,\widehat{\beta}_N)\)</span>.</p>
<p>On estime <span class="math inline">\(\lambda_e\)</span> et <span class="math inline">\(\lambda_c\)</span> en maximisant la vraisemblance.</p>
<section id="ecriture-de-la-vraisemblance" class="level3">
<h3 class="anchored" data-anchor-id="ecriture-de-la-vraisemblance">Ecriture de la vraisemblance</h3>
<p><span class="math display">\[(\widehat{\lambda_e}, \widehat{\lambda_c}) = \arg\max \log \mathcal{L} \left((T_i^R,T_i^V)_{i=1,\dots,n}; \widehat{\theta}_N,\lambda_e,\lambda_c\right)
\]</span></p>
<p>Les temps <span class="math inline">\(T_i^R,T_i^V\)</span> ne sont pas la réalisation de variables aléatoires indépendantes. Cependant leur loi jointe est difficile à calculer. Nous proposons de modifier le critère de la façon suivante:</p>
<p><span class="math display">\[ (\widehat{\lambda_e}, \widehat{\lambda_c}) = \arg\max \log \mathcal{L} \left((T_i^R)_{i=1,\dots,n}; \widehat{\theta}_N,\lambda_e,\lambda_c\right) + \log \mathcal{L} \left((T_i^V)_{i=1,\dots,n}; \widehat{\theta}_N,\lambda_e,\lambda_c\right)
\]</span> Il faut maintenant l’expression de chaque terme donc avoir la loi marginale de <span class="math inline">\(T_i^R\)</span> et <span class="math inline">\(T_i^V\)</span>.<br>
<span class="math inline">\(T_i^R\)</span> s’exprime comme le <span class="math inline">\(\min\)</span> de deux quantités. Il faut donc en déduire sa dénsité.</p>
<p><strong>Rappel</strong></p>
<p>Si <span class="math inline">\(T = \min(U,V)\)</span> avec <span class="math inline">\(U \sim f_U\)</span> et <span class="math inline">\(V \sim f_V\)</span> alors</p>
<p><span class="math display">\[
\begin{array}{ccl}
F_{T}(t) &amp;=&amp; 1 - (1-F_{U}(t))(1-F_{V}(t))\\
f_{T}(x)  &amp;=&amp; f_{V}(x) F_{U}(x) + F_{V}(x) f_{U}(x) - f_{V}(x)  - f_{U}(x)
\end{array}
\]</span></p>
<p>Nous avons besoin de <span class="math inline">\((f_{U^R}, F_{U^R})\)</span> et <span class="math inline">\((f_{V^R}, F_{V^R})\)</span>.</p>
<p><span class="math display">\[V^V \sim \mathcal{E}(\lambda_c)+ \Gamma(k,\lambda_e) = \Gamma(1,\lambda_c)+ \Gamma(k,\lambda_e)\]</span></p>
<p>Cette loi n’a pas de densité explicite mais comme somme de loi Gamma on peut l’approcher la façon suivante:</p>
<p><span class="math display">\[ \Gamma(1,\lambda_e)+ \Gamma(k,\lambda_c) \approx \Gamma(k_L,\lambda_L)\]</span> avec</p>
<p><span class="math display">\[
\begin{array}{cclccl}
R &amp;=&amp; \frac{1}{\lambda_c} + \frac{k}{\lambda_e}, &amp;
Q &amp;=&amp; \frac{1}{\lambda_c^2} + \frac{k}{\lambda_e^2}\\
k_L &amp;=&amp; \frac{R^2}{Q},&amp; \lambda_L &amp;=&amp;\frac{k_L}{R}
\end{array}
\]</span></p>
</section>
<section id="illustration-de-lapproximation-de-la-somme-de-2-lois-gamma-par-une-loi-gamma" class="level3">
<h3 class="anchored" data-anchor-id="illustration-de-lapproximation-de-la-somme-de-2-lois-gamma-par-une-loi-gamma">Illustration de l’approximation de la somme de 2 lois Gamma par une loi Gamma</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># paramètres </span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>lambda_e <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span><span class="fl">0.5</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>lambda_c <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">15</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>k <span class="ot">&lt;-</span> <span class="dv">16</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Simulation d'un échantillon</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="dv">10000</span>;</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>VV <span class="ot">&lt;-</span> <span class="fu">rgamma</span>(n,<span class="dv">1</span>,lambda_c) <span class="sc">+</span> <span class="fu">rgamma</span>(n,k,lambda_e)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>VV <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(VV)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Approximation</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>R <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">c</span>(<span class="dv">1</span>,k)<span class="sc">/</span><span class="fu">c</span>(lambda_c,lambda_e))</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>Q <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">c</span>(<span class="dv">1</span>,k)<span class="sc">/</span><span class="fu">c</span>(lambda_c,lambda_e)<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>k_sum <span class="ot">&lt;-</span> R<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span>Q</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>lambda_sum <span class="ot">&lt;-</span> k_sum<span class="sc">/</span>R</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The dot-dot notation (`..density..`) was deprecated in ggplot2 3.4.0.
ℹ Please use `after_stat(density)` instead.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Cinetique_Proteines_files/figure-html/approx%20density-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="Cinetique_Proteines_files/figure-html/approx%20repat-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="fonctions-de-vraisemblance." class="level3">
<h3 class="anchored" data-anchor-id="fonctions-de-vraisemblance.">Fonctions de vraisemblance.</h3>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>